xtensa  xt-xcc -Wno-unused-but-set-variable  not work

**Describe the bug**
xt-xcc -Wno-unused-but-set-variable  not work

**To Reproduce**
Steps to reproduce the behavior:
west build -b qemu_xtensa samples/hello_world


**Logs and console output**
xt-xcc.exe ERROR parsing -Wno-unused-but-set-variable:  unknown flag
ninja: build stopped: subcommand failed.

**Environment (please complete the following information):**
 - OS: (Windows)
 - Toolchain (xt-xcc)



**Additional context**
Add any other context about the problem here.

--- a/cmake/compiler/gcc/compiler_flags.cmake
+++ b/cmake/compiler/gcc/compiler_flags.cmake
@@ -89,7 +89,7 @@ check_set_compiler_property(APPEND PROPERTY warning_dw_3
                             -Wvla
 )
 
-set_compiler_property(PROPERTY warning_extended -Wno-unused-but-set-variable)
+check_set_compiler_property(PROPERTY warning_extended -Wno-unused-but-set-variable)
 
 check_set_compiler_property(PROPERTY warning_error_implicit_int -Werror=implicit-int)
 

